jakarta-eclipse/src/org/jakarta/lsp4e/JakartaLanguageClient.java

```diff
@@ -78,4 +90,19 @@ public boolean isCanceled() {
  			return JDTServicesManager.getInstance().getExistingContextsFromClassPath(uri, snippetContexts);
  		});
  	}
+
+	public CompletableFuture<List<CodeAction>> getCodeAction(CodeActionParams params){
+		Activator.log(new Status(IStatus.INFO, "code action request received", "code action request receieved"));
+		IProgressMonitor monitor = new NullProgressMonitor();
+		JDTUtils utils = new JDTUtils();
+		JakartaJavaCodeActionParams JakartaParams = new JakartaJavaCodeActionParams(params.getTextDocument(), params.getRange(), params.getContext());
+		return CompletableFutures.computeAsync((cancelChecker) -> {
+			try {
```

Comment:
Instead of a `NullProgressMonitor`, use the one provided by the `cancelChecker`:
```
IProgressMonitor monitor = getProgressMonitor(cancelChecker);
```